### ISSUE TITLE
[Question] Does SB3 provide a possibility to use an Ordinal Distribution Network Architecture for PPO?

### ISSUE THREAD

**Issue by kitaird** [DLR-RM/stable-baselines3#838]
### Question

Does SB3 provide a possibility to use an Ordinal Distribution Network Architecture for PPO?
I read about it in this paper [Discretizing Continuous Action Space for On-Policy Optimization](https://arxiv.org/pdf/1901.10500.pdf) and the results were promising for various tasks in Mujoco, rllab and roboschool. Is this already considered in PPO when using a multi-discrete action space? Or has this to be implemented by a custom network architecture?

### Checklist

- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by Miffyli** [comment]
Without reading deeper into this technology: no, there is no support for ordinal distribution network architectures. Continuous actions are handled with continuous distributions. You could bin the range of valid continuous actions into a multi-discrete choice, and use multi-discrete action space on that, but I do not know how that relates to the architecture you mention.

**Comment by araffin** [comment]
> . You could bin the range of valid continuous actions into a multi-discrete choice,

Yes, the easiest way would be to use a Gym wrapper.

**Comment by kitaird** [comment]
But discretizing the continuous actions is not the only change done in the paper. They also made changes to the probability distribution as far as I can tell. They used a factorized distribution for discretized actions, and for the ordinal architecture they passed the output of the logits into another sigmoid function,, did further computation (chapter 4.2 of the paper) and then passed this into softmax.

They also build upon open-ai baselines and provided their code. However, I'm personally not that sound on this ground to migrate their implementation into SB3. It seems that only a custome NetworkArchitecture would be enough (maybe an additional Distribution). The code of the authors can be found here: [Might want to have a look at ppo_ordinal/policies.py](https://github.com/robintyh1/onpolicybaselines/tree/master/onpolicyalgos/discrete_actions_space)

**Comment by araffin** [comment]
You probably would like to take a look at [Is Bang-Bang Control All You Need? Solving Continuous Control with Bernoulli Policies](https://openreview.net/forum?id=9BvDIW6_qxZ) too.

> the results were promising for various tasks in Mujoco, rllab and roboschool

If you want good results on MuJoCo like task (continuous control), you should take a look at SAC/TD3/TQC (cf. doc) and I would say if you spend enough time tuning hyperparameters, there should be almost no difference between those algorithms. 

> It seems that only a custom NetworkArchitecture would be enough (maybe an additional Distribution).

But anyway, I don't know much about the paper, so yes, you probably need a custom network architecture and distribution.
